### PR TITLE
Use Firefox Sync IDS with Syncto.

### PR DESCRIPTION
**Pull request by Natim**
Do not merge until Kinto.js supports Firefox Sync IDS. (See https://github.com/Kinto/kinto.js/issues/115)

r? @michielbdejong

### CODE REVIEW

**Comment by Natim** [comment]
We need to add a Firefox Sync ID validator there.

**Comment by Natim** [comment]
See 
- https://github.com/mozilla-services/server-syncstorage/blob/master/syncstorage/views/__init__.py#L46-L47
- https://github.com/mozilla-services/server-syncstorage/blob/master/syncstorage/bso.py#L23

**Comment by michielbdejong** [comment]
Ah great, my first 'r?' mention :)
I'll try it out as soon as kinto.js can handle Sync id's and then do the review, ok?

**Comment by Natim** [comment]
It works for me.

My general questions are:

Should syncto accept every accepted ids in Firefox Sync (meaning 64 printable characters for record) or should we restrict records ids to look like today's generated ids on Firefox Sync side to prevent a mix between uuid4 and base64 ids in collections?

**Comment by michielbdejong** [comment]
> should we restrict records

Yes, sounds useful to me. But do keep in mind that we also want to retrieve `meta/global` and `crypto/keys` through syncto, so just the "exactly 12 base64-urlsafe characters" id's used for BSO's is too strict. https://github.com/mozilla-services/docs/blob/master/source/storage/apis-1.5.rst#collections

**Comment by Natim** [comment]
> do keep in mind that we also want to retrieve meta/global and crypto/keys through syncto

Yes sure.

**Comment by Natim** [comment]
``` python
SYNC_ID_FORMAT = re.compile(r'^[a-zA-Z0-9_-]{12}$')  # 9 bytes URL safe base64
```

**Comment by Natim** [comment]
This is going to be implemented with pluggable ID generator and validator here: https://github.com/Kinto/kinto.js/issues/138

**Comment by michielbdejong** [comment]
I tried to rebase this branch on master, but could not really work out what's what in the merge conflicts, could you maybe rebase it? (not very urgent, but would be a nice-to-have, to make our testing of the sync app more convenient until it's merged).

I expect to have the Kinto.js side of things sorted some time during the next week, so we can also just wait for that.

**Comment by michielbdejong** [comment]
@natim https://github.com/Kinto/kinto.js/issues/138 is now closed, and I have been using this branch for testing for a while now without problems, so r+

**Comment by Natim** [comment]
Rebased.

**Comment by Natim** [comment]
@michielbdejong Could you make sure my rebase still works with your patch? If yes, I will merge.

**Comment by michielbdejong** [comment]
Yes! Everything seems to work.

I got some intermittent 503 errors, maybe due to my network connection, maybe timeouts are too short? But retrying a few times would eventually give 200s.

Good to merge as far as I'm concerned - thanks a lot!!

File: syncto/views/record.py [hunk 1]
```diff
@@ -53,7 +50,7 @@ def record_get(request):
 def record_put(request):
     collection_name = request.matchdict['collection_name']
     record_id = request.matchdict['record_id']
-    sync_id = uuid4_to_base64(record_id)
+    sync_id = record_id
```
Comment (Natim) [inline syncto/views/record.py:53]: we can probably get rid of these.

**Pull request merged**